### PR TITLE
Bug 1806620: reduce ldap connection timeout from 60s to 30s

### PR DESCRIPTION
**Pull request by vareti**
LDAP connection timeout was 60s. This might end up eating request timeout(60s) if there is an issue connecting to LDAP server causing the user to login failure without even trying other IDPs. As all the other IDP clients seems to be using 30s connection timeout, it is logical to set LDAP connection timeout to 30s too.

### CODE REVIEW

File: pkg/authenticator/password/ldappassword/ldap.go [hunk 1]
```diff
@@ -42,6 +43,7 @@ type Authenticator struct {
 
 // New returns an authenticator which will validate usernames/passwords using LDAP.
 func New(providerName string, options Options, mapper authapi.UserIdentityMapper) (authenticator.Password, error) {
+	ldap.DefaultTimeout = 30 * time.Second
```
Comment (stlaz) [inline pkg/authenticator/password/ldappassword/ldap.go]: Please add a comment about why you're doing this and that it is a global variable of the ldap module that we're writing into.

Comment (vareti) [reply pkg/authenticator/password/ldappassword/ldap.go]: addressed it! please take a look!

**Comment by sttts** [comment]
This needs a release note and documentation cc @bergerhoffer. 

@vareti can you link to a release note change? @bergerhoffer can give you pointers how to do that.

File: pkg/authenticator/password/ldappassword/ldap.go (hunk 1 above)
Comment (sttts) [reply pkg/authenticator/password/ldappassword/ldap.go]: So this means that out of 60 sec request timeout we have 30s left at worst? This still kills every request taking longer than 30s to be processed by etcd.

**Comment by bergerhoffer** [comment]
@sttts @vareti Is this for 4.4 or 4.5? And are you considering this a bug fix, or more of a feature that you can fall back to a different IDP to log in (or was it only in this situation you couldn't fall back)?

File: pkg/authenticator/password/ldappassword/ldap.go (hunk 1 above)
Comment (vareti) [reply pkg/authenticator/password/ldappassword/ldap.go]: Yes, I agree. There are other failure cases too. If there are 3 IDPs configured(assuming it is a realistic scenario) and if the first two IDPs take 30s, request would be killed without trying the third IDP. 

I am not sure what repercussions increasing the request timeout would have. I didn't feel comfortable increasing it and so i ended up reducing the ldap timeout. Do you think it is better or there is a better way of solving it?

**Comment by vareti** [comment]
@bergerhoffer This is a 4.4 bug fix. Fall back to different IDP is the expectation but the behavior seems to be different.

**Comment by bergerhoffer** [comment]
Great thanks @vareti! I'll make a note of it in the release notes tracker just so we ensure it gets added, but what would help is if you could add a summary of the issue/resolution in the **Doc Text** field of the BZ. Thanks!

**Comment by bergerhoffer** [comment]
@vareti Also looking at the BZ, it looks like the target release is 4.5. Should it be 4.4 if it's a bug fix for 4.4? Just wanting to make sure it shows up when we do our query for all 4.4 bug fixes.

**Comment by sttts** [comment]
/lgtm
/approve

This is a little improvement. Not a real solution. At least the cluster does not break down completely.

**Pull request merged**